lightningd/dual_open_control.c

```diff
@@ -645,13 +645,23 @@ openchannel2_hook_cb(struct openchannel2_payload *payload STEALS)
 		return subd_send_msg(dualopend, take(msg));
 	}
 
+	// Determine the wallet index for our upfront_shutdown_script, UINT32_MAX if not found.
+	u32 our_shutdown_script_wallet_index = UINT32_MAX;
+	bool is_p2sh;
+	wallet_can_spend(
+		dualopend->ld->wallet,
+		payload->our_shutdown_scriptpubkey,
+		&our_shutdown_script_wallet_index,
+		&is_p2sh);
```

Comment:
Relying on internals of wallet_can_spend is a bad idea: if it returns False, set our_shutdown_script_wallet_index to UINT32_MAX.

Reply:
Actually, even better, you can make our_shutdown_script_wallet_index an optional parameter in the wire protocol (using the notation `?u32`) which makes it unambiguous vs a magic value?

Reply:
fixed